docs/concepts.md

```diff
@@ -51,8 +58,325 @@ The newly provisioned resource is automatically bound to the resource claim.
+The registry understands a Stack’s custom controller and its CRDs and indexes by both -- you could lookup a custom controller by the CRD name and vice versa.
+
+### Stack Package Format
+The package format for Stacks that contains the Stack definition, metadata, icons, CRDs, and other Stack specific files.  
+
+### Stack Manager
+The component that is responsible for installing a Stack’s custom controllers and resources in Crossplane. It can download packages, resolve dependencies, install resources and execute controllers.  This component is also responsible for managing the complete life-cycle of Stacks, including upgrading them as new versions become available.
+
+### External Resource
+An actual resource that exists outside Kubernetes, typically in the cloud. 
+AWS RDS and GCP Cloud Memorystore instances are external resources.
+
+### Managed Resource
+The Crossplane representation of an external resource.
+The `RDSInstance` and `CloudMemorystoreInstance` Kubernetes kinds are managed
+resources. A managed resource models the satisfaction of a need; i.e. the need
+for a Redis Cluster is satisfied by the allocation (aka binding) of a
+`CloudMemoryStoreInstance`.
+
+### Resource Claim
+The Crossplane representation of a request for the
+allocation of a managed resource. Resource claims typically represent the need
+for a managed resource that implements a particular protocol. `MySQLInstance`
+and `RedisCluster` are examples of resource claims.
+
+### Resource Class
+The Crossplane representation of the desired configuration
+of a managed resource. Resource claims reference a resource class in order to
+specify how they should be satisfied by a managed resource.
+
+### Cloud-Specific Resource Class
+Cloud-specific resource classes define configuration for a specific type of
+service that is offered by a cloud provider.
+* aka non-portable class
+
+### Non-Portable Resource Class
+See cloud-specific resource classs.
+
+### Portable Resource Class
+Offers portable classes of service in Kubernetes namespace
+
+### Provider
+Cloud service provider configured into used to provision
+managed services in Crossplane
+
+### Infrastructure Namespace
+Kubernetes namespace in the Crossplane Instance where 
+
+### Project Namespace
+see App Project Namespace
+
+### App Project Namespace
+TBD
+
+### Application Stack
+TBD
+
+### Infrastructure Stack
+TBD
+
+### Cloud Provider Stack
+TBD
+
+### Connection Secret
+A Kubernetes `Secret` encoding all data required to
+connect to (or consume) an external resource.
+
+### Claimant
+The Kubernetes representation of a process wishing
+to connect to a managed resource, typically a `Pod` or some abstraction
+thereupon such as a `Deployment` or `KubernetesApplication`.
+
+### Consumer
+See claimant.
+
+### Claim Kind
+The specific kind of claim, for example a MySQLInstance claim.
+
+### Kubernetes Application
+* aka Complex Workload
```

Comment:
Technically Kubernetes Application is a kind of complex workload. We named it such to allow space for e.g. `kubernetesapplication.workload.crossplane.io`, `vmapplication.workload.crossplane.io`, `function.workload.crossplane.io`, etc.

Reply:
fixed